src/Development/Shake/Command.hs

```diff
@@ -140,34 +140,35 @@ commandExplicit funcName copts results exe args = do
             [] -> traced (takeFileName exe)
 
     let tracker act = case shakeLint opts of
-            Just LintTracker -> (if isWindows then winTracker else unixTracker) act
-            _ -> act [] exe args
+            Just LintTracker -> winTracker act
+            Just LintFSATrace -> fsatrace act
+            _ -> act exe args
 
         winTracker act = do
             (dir, cleanup) <- liftIO newTempDir
             flip actionFinally cleanup $ do
-                res <- act [] "tracker" $ "/if":dir:"/c":exe:args
+                res <- act "tracker" $ "/if":dir:"/c":exe:args
                 (rs, ws) <- liftIO $ trackerFiles dir
                 trackRead rs
                 trackWrite ws
                 return res
 
-        unixTracker act = do
+        fsatrace act = do
             (file, cleanup) <- liftIO newTempFile
             flip actionFinally cleanup $ do
                 fsat <- liftIO $ getEnv "FSAT"
-                let vars = [AddEnv "DYLD_INSERT_LIBRARIES" fsat
-                           ,AddEnv "DYLD_FORCE_FLAT_NAMESPACE" "1"
-                           ,AddEnv "FSAT_OUT" file]
-                res <- act vars exe args
+                res <- act fsat $ file:"--":exe:args
                 (rs, ws) <- liftIO $ fsatraceFiles file
-                whitelist <- liftIO unixWhitelist
-                let whitelisted x = any (\w -> (w ++ "/") `isPrefixOf` x) whitelist
-                trackRead $ filter (not . whitelisted) rs
-                trackWrite $ filter (not . whitelisted) ws
+                let whitelisted x = any (`isPrefixOf` x) unixWhitelist
+                    ham = filter (not . whitelisted)
+                    rel = mapM (liftIO . makeRelativeToCurrentDirectory)
```

Comment:
`makeRelativeToCurrentDirectory` is literally defined as `makeRelative` on `getCurrentDirectory`, so I would suggest pulling out `getCurrentDirectory` once, and then calling `makeRelative` many times, which makes this step pure and likely faster.